### DISCUSSION TITLE
Refactoring and verification of heatmap based algorithms on DeepFashion dataset DeepFashion 数据集 Heatmap 算法迁移

### DISCUSSION THREAD

**Discussion by OpenMMLab-Assistant-004** [open-mmlab/OpenMMLabCamp#533] [MMPose]
# Refactoring and verification of heatmap based algorithms on DeepFashion dataset 
 DeepFashion 数据集 Heatmap 算法迁移 

Migrate Heatmap algorithm on [DeepFashion dataset from 0.x branch]([https://github.com/open-mmlab/mmpose/tree/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion)](https://github.com/open-mmlab/mmpose/tree/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion))  into dev-1.x branch. Config is [here]([https://github.com/open-mmlab/mmpose/blob/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion/res50_deepfashion_full_256x192.py)](https://github.com/open-mmlab/mmpose/blob/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion/res50_deepfashion_full_256x192.py)) 
 . Make sure MMPose 1.0 can load the old version ckpts and achieve compariable accuracy(within 0.1AP). If the accuracy cannot match, feel free to contact us.
 1. Run test script to make sure the inference accuracy is the same as it in 0.x. (Better to try training as well, and keep the training ckpt and log file)
 2. If there exists miss matching keys in ckpts, refer to [doc]([https://mmpose.readthedocs.io/en/latest/migration.html)](https://mmpose.readthedocs.io/en/latest/migration.html))  to migrate.
 3. Create the new config under configs/fashion_2d_keypoint/topdown_heatmap/deep_fashion/ , and name it following [doc]([https://mmpose.readthedocs.io/en/latest/user_guides/configs.html#config-file-naming-convention)](https://mmpose.readthedocs.io/en/latest/user_guides/configs.html#config-file-naming-convention)) 
 4. You can refer to config of DeepFashion2 to create your new config of DeepFashion.[like this]([https://github.com/open-mmlab/mmpose/tree/main/configs/fashion_2d_keypoint/topdown_heatmap/deepfashion2)](https://github.com/open-mmlab/mmpose/tree/main/configs/fashion_2d_keypoint/topdown_heatmap/deepfashion2)) 

将 [mmpose 0.x 分支]([https://github.com/open-mmlab/mmpose/tree/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion)](https://github.com/open-mmlab/mmpose/tree/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion))  上 [DeepFashion 支持的 Heatmap 算法]([https://github.com/open-mmlab/mmpose/blob/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion/res50_deepfashion_full_256x192.py)](https://github.com/open-mmlab/mmpose/blob/0.x/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion/res50_deepfashion_full_256x192.py))  config 迁移到 dev-1.x 分支，并进行 ckpt 兼容性测试和推理精度对齐，确保在 MMPose 1.0 中可以加载旧版 ckpt 并达到同等推理精度（误差0.1AP以内），如果存在精度对不上的情况可以开issue和我们讨论一下。
 1. 请测试代码可用性，确认代码可以顺利跑通训练流程，不必完成训练（如果有能力完成是更好的）。
 2. 新旧ckpt权重文件中如果存在 key不兼容问题，ckpt兼容方法参考 [doc]([https://mmpose.readthedocs.io/en/latest/migration.html)](https://mmpose.readthedocs.io/en/latest/migration.html)) 
 3. 新的 config 请存放到 configs/face_2d_keypoint/topdown_regression/wflw/ 下，config 命名规则参考 [doc]([https://mmpose.readthedocs.io/en/latest/user_guides/configs.html#config-file-naming-convention)](https://mmpose.readthedocs.io/en/latest/user_guides/configs.html#config-file-naming-convention)) 
 4. 新版的config可以参考 deepfashion2 中 topdown_heatmap下的已迁移的 config ，如 [page]([https://github.com/open-mmlab/mmpose/tree/main/configs/fashion_2d_keypoint/topdown_heatmap/deepfashion2)](https://github.com/open-mmlab/mmpose/tree/main/configs/fashion_2d_keypoint/topdown_heatmap/deepfashion2)) 

## Task Information

**Repository**：MMPose

**Difficulty**：Medium

**Points**：30

**Estimated Time Consumption**：14 days

## Task Status

**Claimant's Github ID**：xin-li-67

**Claim Time**：2023年08月03日

**PR Link**：https://github.com/open-mmlab/mmpose/pull/2597

## Points to Note

❗️请注意：此任务属于超级视客营任务，不能被领取！请登陆 [超级视客营平台](https://openmmlab.com/activity/codecamp) 按照流程申请任务，我们会尽快审核～

**Comment by xin-li-67** [comment]
This [PR#2597](https://github.com/open-mmlab/mmpose/pull/2597) is under construction ...

**Reply by crazysteeaam** [reply]
Could you please prefix the PR title starting with [CodeCamp2023-533]? Thank you!

**Reply by xin-li-67** [reply]
Done~
